service-worker/bootstrap.js

```diff
@@ -249,6 +245,27 @@ self.addEventListener('install', event => {
     })
 })
 
+self.addEventListener('fetch', event => {
+  // Catches all non-prefetch requests and aborts in-progress prefetches
+  // until the request finishes, then resumes prefetching
+  abortPrefetches()
+  event.respondWith(
+    fetch(event.request)
+      .then(resp => {
+        if (toResume.size) {
+          resumePrefetches()
+        }
+        return resp
+      })
+      .catch(e => {
+        if (toResume.size) {
+          resumePrefetches()
```

Comment:
I guess resumePrefetches can be moved to finally, then there would be no code duplicatigon

Reply:
In order to do that, we would need to rewrite this using async/await instead of promises, which we should, as I believe those are natively supported in the service worker.

Reply:
Done.